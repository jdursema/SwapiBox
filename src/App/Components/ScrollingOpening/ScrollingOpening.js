import React from 'react';
import './ScrollingOpening.css';
import PropTypes from 'prop-types';

const OpeningScrolling = ({ OpeningCrawl }) => {
  return (
    <div className='scrolling-opening'>
      <div className='crawl'>
        <div className='fade'></div>
        <h1 className='scrolling-opening-title'>{OpeningCrawl.title}</h1>
        <div>
          {OpeningCrawl.body &&
          <div>
            <p>{OpeningCrawl.body[0]}</p>
            <p>{OpeningCrawl.body[1]}</p>
            <p>{OpeningCrawl.body[2]}</p>
          </div>}

        </div>
      </div>
    </div>
  );
};

export default OpeningScrolling;

OpeningScrolling.propTypes = {
  OpeningCrawl: PropTypes.object
};