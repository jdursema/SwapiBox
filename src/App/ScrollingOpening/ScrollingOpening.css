.scrolling-opening{
  position: fixed;
  background-image: url(stars.png);
  background-repeat: no-repeat;
  background-size: cover;
  z-index: 10;
  display: flex;
  display: flex;
  align-self: center;
  border: yellow 3px solid;
  height: 800px;
  width: 400px;
  justify-content: center;
}

.crawl{
  position: relative;
  top: -100px;
  transform-origin: 50% 100%;
  animation: crawl 60s linear;
}

@keyframes crawl {
  0% {
    top: 0;
    transform: rotateX(20deg) translateZ(0);
  }
  100% { 
    top: -6000px;
    transform: rotateX(25deg) translateZ(-2500px);
  }
}

.fade {
  position: relative;
  width: 100%;
  min-height: 60vh;
  top: -25px;
  background-image: linear-gradient(0deg, transparent, black 75%);
  z-index: 1;
}

.scrolling-opening-text{
  color: white;
}

.scrolling-opening-title{
  color: white;
}
