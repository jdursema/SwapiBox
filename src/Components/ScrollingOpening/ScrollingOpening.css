.scrolling-opening{
  background-color: black;
  background-repeat: no-repeat;
  background-size: cover;
  z-index: 10;
  border: yellow 10px solid;
  height: 98%;
  width: 98%;
  position: fixed;
}

.close-btn{
  background-color:red;
  color:white;
  display: flex;
  justify-content: left;
}

.crawl{
  position: relative;
  top: -100px;
  transform-origin: 50% 100%;
  animation: crawl 60s linear;
}

@keyframes crawl {
	0% { top: 100%; }
	100% { top: -170%; }
}

.fade {
  position: relative;
  width: 80%;
  min-height: 60vh;
  top: -25px;
  background-image: linear-gradient(0deg, transparent, black 75%);
  z-index: 1;
}

.scrolling-opening-text{
  color: white;
}

.scrolling-opening-title{
  color: white;
  font-size: 75px;
  font-family: 'News Gothic Std', sans-serif;
}

p{
  color: white;
  font-size: 30px;
  margin: 5%;
  font-family: 'News Gothic Std', sans-serif;
}

.hide-screen{
  display: none;
}


