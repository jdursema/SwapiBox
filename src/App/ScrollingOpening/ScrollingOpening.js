import React from 'react';
import './ScrollingOpening.css';

const OpeningScrolling = ({OpeningCrawl}) => {
  return(
    <div className='scrolling-opening'>
      <div className='crawl'>
      <div className='fade'></div>
      <h1 className='scrolling-opening-title'>{OpeningCrawl.title}</h1>
       <p className='scrolling-opening-text'>{OpeningCrawl.body}</p>
      </div>
    </div>
  )
}

export default OpeningScrolling;