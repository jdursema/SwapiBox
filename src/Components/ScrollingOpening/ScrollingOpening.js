import React, {Component} from 'react';
import './ScrollingOpening.css';
import PropTypes from 'prop-types';

class OpeningScrolling extends Component {
  constructor (props) {
    super();
    this.state = {
      display: 'scrolling-opening'
    };
  }

  closeScreen = () => {
    this.setState({display: 'scrolling-opening hide-screen'});
  }

  render () {
    return (
      <div className={this.state.display}>
        <button className='close-btn' onClick={ ()=>{ this.closeScreen(); } }>
          X
        </button>
        <div className='crawl'>
          <h1 className='scrolling-opening-title'>
            {this.props.OpeningCrawl.title}
          </h1>
          <div>
            {this.props.OpeningCrawl.body &&
            <div>
              <p>{this.props.OpeningCrawl.body[0]}</p>
              <p>{this.props.OpeningCrawl.body[1]}</p>
              <p>{this.props.OpeningCrawl.body[2]}</p>
            </div>}
            <p>{this.props.OpeningCrawl.releaseDate}</p>
          </div>
        </div> 
      </div>
    );
  } 
}

export default OpeningScrolling;

OpeningScrolling.propTypes = {
  OpeningCrawl: PropTypes.object
};