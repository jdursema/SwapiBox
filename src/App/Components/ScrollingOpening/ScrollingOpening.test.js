import React from 'react';
import { shallow } from 'enzyme';
import ScrollingOpening from './ScrollingOpening';

it('renders without crashing', () => {
  const openingCrawl = {title: 'Star Wars!', body: ['Yogurt', 'I hate', 'Yogurt'], releaseDate: 'Jan 1'};
  const scrollingOpening = shallow(<ScrollingOpening OpeningCrawl={openingCrawl}/>);

  expect(scrollingOpening).toBeDefined();
})

it('should match its snapshot', () => {
  const openingCrawl = {title: 'Star Wars!', body: ['Yogurt', 'I hate', 'Yogurt'], releaseDate: 'Jan 1'};
  const scrollingOpening = shallow(<ScrollingOpening OpeningCrawl={openingCrawl}/>);
  
  expect(scrollingOpening).toMatchSnapshot();
});