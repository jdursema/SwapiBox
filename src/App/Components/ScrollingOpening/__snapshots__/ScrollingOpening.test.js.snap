// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match its snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OpeningScrolling
    OpeningCrawl={
        Object {
            "body": Array [
              "Yogurt",
              "I hate",
              "Yogurt",
            ],
            "releaseDate": "Jan 1",
            "title": "Star Wars!",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="crawl"
>
        <div
                className="fade"
        />
        <h1
                className="scrolling-opening-title"
        >
                Star Wars!
        </h1>
        <div>
                <div>
                        <p>
                                Yogurt
                        </p>
                        <p>
                                I hate
                        </p>
                        <p>
                                Yogurt
                        </p>
                </div>
                <p>
                        Jan 1
                </p>
        </div>
</div>,
      "className": "scrolling-opening",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="fade"
/>,
          <h1
            className="scrolling-opening-title"
>
            Star Wars!
</h1>,
          <div>
            <div>
                        <p>
                                    Yogurt
                        </p>
                        <p>
                                    I hate
                        </p>
                        <p>
                                    Yogurt
                        </p>
            </div>
            <p>
                        Jan 1
            </p>
</div>,
        ],
        "className": "crawl",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fade",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Star Wars!",
            "className": "scrolling-opening-title",
          },
          "ref": null,
          "rendered": "Star Wars!",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <p>
                                Yogurt
                </p>
                <p>
                                I hate
                </p>
                <p>
                                Yogurt
                </p>
</div>,
              <p>
                Jan 1
</p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Yogurt
</p>,
                  <p>
                    I hate
</p>,
                  <p>
                    Yogurt
</p>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Yogurt",
                  },
                  "ref": null,
                  "rendered": "Yogurt",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "I hate",
                  },
                  "ref": null,
                  "rendered": "I hate",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Yogurt",
                  },
                  "ref": null,
                  "rendered": "Yogurt",
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Jan 1",
              },
              "ref": null,
              "rendered": "Jan 1",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="crawl"
>
          <div
                    className="fade"
          />
          <h1
                    className="scrolling-opening-title"
          >
                    Star Wars!
          </h1>
          <div>
                    <div>
                              <p>
                                        Yogurt
                              </p>
                              <p>
                                        I hate
                              </p>
                              <p>
                                        Yogurt
                              </p>
                    </div>
                    <p>
                              Jan 1
                    </p>
          </div>
</div>,
        "className": "scrolling-opening",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="fade"
/>,
            <h1
              className="scrolling-opening-title"
>
              Star Wars!
</h1>,
            <div>
              <div>
                            <p>
                                          Yogurt
                            </p>
                            <p>
                                          I hate
                            </p>
                            <p>
                                          Yogurt
                            </p>
              </div>
              <p>
                            Jan 1
              </p>
</div>,
          ],
          "className": "crawl",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fade",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Star Wars!",
              "className": "scrolling-opening-title",
            },
            "ref": null,
            "rendered": "Star Wars!",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div>
                  <p>
                                    Yogurt
                  </p>
                  <p>
                                    I hate
                  </p>
                  <p>
                                    Yogurt
                  </p>
</div>,
                <p>
                  Jan 1
</p>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p>
                      Yogurt
</p>,
                    <p>
                      I hate
</p>,
                    <p>
                      Yogurt
</p>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Yogurt",
                    },
                    "ref": null,
                    "rendered": "Yogurt",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "I hate",
                    },
                    "ref": null,
                    "rendered": "I hate",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Yogurt",
                    },
                    "ref": null,
                    "rendered": "Yogurt",
                    "type": "p",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Jan 1",
                },
                "ref": null,
                "rendered": "Jan 1",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
